api/src/main/java/com/foro/api/service/TopicReplyService.java

```diff
@@ -0,0 +1,125 @@
+import com.foro.api.topic.TopicDTO;
+import com.foro.api.topic.TopicRepository;
+import com.foro.api.topic.TopicStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+@Service
+public class TopicReplyService {
+    @Autowired
+    private TopicRepository topicRepository;
+    @Autowired
+    private ReplyRepository replyRepository;
+
+    @Transactional(readOnly = true)
+    public List<Topic> topicList() {
+        List<TopicDTO> topicDTOList = topicRepository.findAll();
+        List<Topic> topicList = topicDTOList.stream().map(topicDTO -> Topic.from(topicDTO)).collect(Collectors.toList());
+        return topicList;
+    }
+
+    @Transactional(readOnly = true)
+    public Topic topicById(Integer id) {
+        TopicDTO topicDTO = topicRepository.findById(id).orElse(null);
+        Topic topicById = Topic.from(topicDTO);
+        return topicById;
+    }
+
+    @Transactional
+    public Topic saveTopic(Topic topic) {
+        validateTopic(topic);
+        TopicDTO topicToSave=new TopicDTO();
+        topicToSave.setTitle(topic.getTitle());
+        topicToSave.setIdUser(topic.getIdUser());
+        topicToSave.setMessage(topic.getMessage());
+        topicToSave.setCourse(topic.getCourse());
+        topicToSave.setCreationDate(LocalDateTime.now());
+        topicToSave.setDeleted(false);
+        topicToSave.setTopicStatus(TopicStatus.OPEN);
+        TopicDTO topicSaved = topicRepository.save(topicToSave);
+        return Topic.from(topicSaved);
+    }
+
+    @Transactional(readOnly = true)
+    public List<Reply> replyList() {
+        List<ReplyDTO> replyDTOList=replyRepository.findAll();
+        List<Reply> replyList=replyDTOList.stream().map(replyDTO -> Reply.fromReply(replyDTO)).collect(Collectors.toList());
+        return replyList;
+    }
+
+    @Transactional(readOnly = true)
+    public Reply replyById(Integer id) {
+        ReplyDTO replyDTO = replyRepository.findById(id).orElse(null);
+        Reply replyById = Reply.fromReply(replyDTO);
+        return replyById;
+    }
+
+    @Transactional
+    public Reply saveReply(Reply reply) {
+        validateReply(reply);
+        ReplyDTO replyDTO=new ReplyDTO();
+        replyDTO.setReply(reply.getReply());
+        replyDTO.setIdUser(reply.getIdUser());
+        replyDTO.setIdTopic(reply.getIdTopic());
+        replyDTO.setCreationDateReply(LocalDateTime.now());
+        replyDTO.setDeletedReply(false);
+        ReplyDTO replySaved = replyRepository.save(replyDTO);
+        return Reply.fromReply(replySaved);
+    }
+
+    @Transactional
+    public boolean deleteReply(Reply reply) {
+        validateReply(reply);
+        ReplyDTO replyToDeleteById = replyRepository.getReferenceById(reply.getId());
+        replyToDeleteById.setDeletedReply(true);
```

Comment:
Since we're doing a logical delete, we will need to update the findById and findAll to filter out the deleted=true, since after deleted they will still be returned, but let's address this in the following MR.